### ISSUE TITLE
[mysqldef] Index prefix length fails to be parsed

### ISSUE THREAD

**Issue by shiroemons** [k0kubun/sqldef#118]
## --export output

```sql
CREATE TABLE `test` (
  `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
  `hoge` text COLLATE utf8mb4_bin NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;
```

## Input SQL

```sql.diff
CREATE TABLE `test` (
  `id` bigint(20) UNSIGNED NOT NULL AUTO_INCREMENT,
  `hoge` TEXT NOT NULL,
  PRIMARY KEY (`id`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE utf8mb4_bin;

+ ALTER TABLE `test` ADD index idx_test_hoge(hoge(255));
```

## Current output

```
found syntax error when parsing DDL "ALTER TABLE `test` ADD index idx_test_hoge(hoge(255))": syntax error at position 49
```

## Expected output

```
-- Apply --
ALTER TABLE `test` ADD index idx_test_hoge(hoge(255));
```

**Comment by shiroemons** [comment]
Is this the same problem?

## --export output

```sql
CREATE TABLE `test` (
  `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
  `hoge` varchar(255) COLLATE utf8mb4_bin NOT NULL,
  PRIMARY KEY (`id`),
  KEY `idx_test_hoge` (`hoge`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin;
```

## Input SQL

```sql
CREATE TABLE `test` (
  `id` bigint(20) UNSIGNED NOT NULL AUTO_INCREMENT,
  `hoge` TEXT NOT NULL,
  PRIMARY KEY (`id`),
  INDEX idx_test_hoge(hoge(255))
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE utf8mb4_bin;
```

## Current output

```
-- Apply --
ALTER TABLE `test` CHANGE COLUMN `hoge` `hoge` text NOT NULL;
2021/05/24 18:19:53 Error 1170: BLOB/TEXT column 'hoge' used in key specification without a key length
```

**Comment by k0kubun** [comment]
I fixed both issues and started the release as v0.10.1.

**Comment by shiroemons** [comment]
Thanks for the quick fix.
